test/TestUtils/TestProjectFixture.cs

```diff
@@ -0,0 +1,362 @@
+            _sharedLibraryPrefix = sharedLibraryPrefix 
+                ?? RuntimeInformationExtensions.GetSharedLibraryPrefixForCurrentPlatform();
+
+            _repoDirectoriesProvider = repoDirectoriesProvider;
+
+            _testProjectSourceDirectory = testProjectSourceDirectory
+                ?? Path.Combine(repoDirectoriesProvider.RepoRoot, "TestAssets", "TestProjects");
+            _testArtifactDirectory = _testArtifactDirectory
+                ?? Environment.GetEnvironmentVariable(s_testArtifactDirectoryEnvironmentVariable)
+                ?? Path.Combine(AppContext.BaseDirectory, s_testArtifactDirectoryEnvironmentVariable);
+
+            _sdkDotnet = new DotNetCli(dotnetInstallPath ?? DotNetCli.GetStage0Path(repoDirectoriesProvider.RepoRoot));
+            _currentRid = currentRid ?? _sdkDotnet.GetRuntimeId();
+            
+            _builtDotnet = new DotNetCli(repoDirectoriesProvider.BuiltDotnet);
+
+            InitializeTestProject(
+                _testProjectName,
+                _testProjectSourceDirectory, 
+                _testArtifactDirectory,
+                _exeExtension,
+                _sharedLibraryExtension,
+                _sharedLibraryPrefix);
+        }
+
+        public TestProjectFixture(TestProjectFixture fixtureToCopy)
+        {
+            _testProjectName = fixtureToCopy._testProjectName;
+            _exeExtension = fixtureToCopy._exeExtension;
+            _sharedLibraryExtension = fixtureToCopy._sharedLibraryExtension;
+            _sharedLibraryPrefix = fixtureToCopy._sharedLibraryPrefix;
+            _repoDirectoriesProvider = fixtureToCopy._repoDirectoriesProvider;
+            _testProjectSourceDirectory = fixtureToCopy._testProjectSourceDirectory;
+            _testArtifactDirectory = fixtureToCopy._testArtifactDirectory;
+            _sdkDotnet = fixtureToCopy._sdkDotnet;
+            _currentRid = fixtureToCopy._currentRid;
+            _builtDotnet = fixtureToCopy._builtDotnet;
+            _sourceTestProject = fixtureToCopy._sourceTestProject;
+
+            _testProject = CopyTestProject(
+                fixtureToCopy.TestProject, 
+                _testArtifactDirectory, 
+                _exeExtension,
+                _sharedLibraryExtension,
+                _sharedLibraryPrefix);
+        }
+
+        private void InitializeTestProject(
+            string testProjectName,
+            string testProjectSourceDirectory,
+            string testArtifactDirectory,
+            string exeExtension,
+            string sharedLibraryExtension,
+            string sharedLibraryPrefix)
+        {
+            var sourceTestProjectPath = Path.Combine(testProjectSourceDirectory, testProjectName);
+            _sourceTestProject = new TestProject(
+                sourceTestProjectPath,
+                exeExtension,
+                sharedLibraryExtension,
+                sharedLibraryPrefix);
+
+            _testProject = CopyTestProject(
```

Comment:
How many times do we copy the test project?  We copy them all in TestTargets.CopyTestProjectsToArtifacts.  And then we copy them here again.  Can't we trim this down to only needing to copy them 1 time?  The test can copy it from the RepoRoot/TestAssets directory to where it needs to run it